main.tf

```diff
@@ -1,53 +1,53 @@
 locals {
-    account_tier              = (var.account_kind == "FileStorage" ? "Premium" : split("_", var.sku)[0])
-    account_replication_type  = (local.account_tier == "Premium" ? "LRS" : split("_", var.sku)[1])
-    resource_group_name = element(
-    coalescelist(data.azurerm_resource_group.rgrp.*.name, azurerm_resource_group.rg.*.name, [""]), 0)
-    location = element(
-    coalescelist(data.azurerm_resource_group.rgrp.*.location, azurerm_resource_group.rg.*.location, [""]), 0)
-    if_static_website_enabled = var.enable_static_website ? [{}] : [] 
+  account_tier             = (var.account_kind == "FileStorage" ? "Premium" : split("_", var.sku)[0])
+  account_replication_type = (local.account_tier == "Premium" ? "LRS" : split("_", var.sku)[1])
+  resource_group_name = element(
+  coalescelist(data.azurerm_resource_group.rgrp.*.name, azurerm_resource_group.rg.*.name, [""]), 0)
+  location = element(
+  coalescelist(data.azurerm_resource_group.rgrp.*.location, azurerm_resource_group.rg.*.location, [""]), 0)
+  if_static_website_enabled = var.enable_static_website ? [{}] : []
 }
 
 data "azurerm_resource_group" "rgrp" {
-    count                     = var.create_resource_group == false ? 1 : 0
-    name                      = var.resource_group_name
+  count = var.create_resource_group ? 0 : 1
+  name  = var.resource_group_name
 }
 
 resource "azurerm_resource_group" "rg" {
-    count                     = var.create_resource_group ? 1 : 0
-    name                      = var.resource_group_name
-    location                  = var.location
-    tags                      = merge({"Name" = format("%s", var.resource_group_name)}, var.tags,)
+  count    = var.create_resource_group ? 1 : 0

```

Comment:
Is this `count = var.create_resource_group ? 0 : 1` logic supposed to be different from the rgrp section above?

Reply:
yes, It should be.  The logic is supposed to invoke the data source when the group is not created.